mitodl/micromasters
# Hid final exam grades that aren't available yet (#3194)

#### What are the relevant tickets?
Fixes #3194

#### What's this PR do?
Adds fields to `ExamRun` and `ProctoredExamGrade` so we can filter out grades that have been processed but aren't available yet.

#### How should this be manually tested?
- Before running the migrations ensure you user has a `ProctoredExamGrade` for an `ExamRun` in the past that has `exam_series_code='PILOT'`. 
- After the migrations the grade should have `exam_run` populated and the run should have `date_grades_available` populated from the value of `date_last_eligible`.
- Ensure you now can still see the grade on the dashboard.
- Change `date_grades_available` to the future and verify you no longer see the grade.

#### Any background context you want to provide?
Since we're constructing a new FK, I didn't set `ProctoredExamGrade.exam_run` `null=False` because I would rather not the migration fail in case we encounter something unexpected. We can reconcile those with issue later if they come up.
